fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -335,6 +336,9 @@ def __init__(
         if self.fp32_reduce_scatter and not self.mixed_precision:
             raise ValueError("fp32_reduce_scatter requires mixed_precision=True")
 
+        if self.ssd_offload and not self.flatten_parameters:
+            raise ValueError("ssd_offload requires flatten_parameters=True")
```

Comment:
Error message should say `OffloadConfig`'s offload type. since ssd_offload is specific to the implementation